src/dsp/pdr.c

```diff
@@ -346,6 +496,47 @@ const pldm_pdr_record *pldm_pdr_fru_record_set_find_by_rsi(
 	return NULL;
 }
 
+LIBPLDM_ABI_STABLE
+const pldm_pdr_record *pldm_pdr_fru_record_find_by_rsi(
+	const pldm_pdr *repo, uint16_t fru_rsi, uint16_t *terminus_handle,
+	uint16_t *entity_type, uint16_t *entity_instance_num,
+	uint16_t *container_id, bool is_remote)
+{
+	if (!repo || !terminus_handle || !entity_type || !entity_instance_num ||
+	    !container_id) {
+		return NULL;
+	}
+
+	uint8_t *data = NULL;
+	uint32_t size = 0;
+	const pldm_pdr_record *curr_record = pldm_pdr_find_record_by_type(
```

Comment:
can we make use of msgbuf concept here also?

Reply:
how do we use msgbuf here , can u please elaborate? this function has used msgbuf when we meed to get struct fields or add them into a struct . other than that there is no scope to use msgbuf for this line

Reply:
i can add msgbuf for the entire api

Reply:
are we planning to do this?

Reply:
No this addressed in subsequent PRs , they also need to added in upstream so I will do both together

Reply:
okay sure